### PR TITLE
homework/lesson-7.2

### PR DESCRIPTION
**Pull request by nguyenvanbapc66**

### CODE REVIEW

**Comment by quanghoang33** [comment]
algo part lgtm

File: homework/ba-nguyen/lesson-7.2/javascript-3.md [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+type UpdateFuncType = (n: number) => number;
+type BodyFuncType = (n: number) => void;
+
+function loop(
+  value: number,
+  testFunc: TestFuncType,
+  updateFunc: UpdateFuncType,
+  bodyFunc: BodyFuncType
+) {
+  while (testFunc(value)) {
```
Comment (thaycacac) [inline homework/ba-nguyen/lesson-7.2/javascript-3.md:76]: this question require use higher-order function

File: homework/ba-nguyen/lesson-7.2/javascript-3.md [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    return (
+      keys1.length === keys2.length &&
+      keys1.every((item, index) => deepEqual(item, keys2[index]))
+    );
+  }
+
+  return value1 === value2;
+}
+
+console.log(deepEqual([1, 2], [1, 2]));
```
Comment (thaycacac) [inline homework/ba-nguyen/lesson-7.2/javascript-3.md:120]: some test case not pass

```js
const obj = {here: {is: "an", other: "3"}, object: 2};
console.log(deepEqual(obj, obj));
// → true
console.log(deepEqual(obj, {here: 1, object: 2}));
// → false
console.log(deepEqual(obj, {here: {is: "an"}, object: 2}));
// → false
console.log(deepEqual(obj, {here: {is: "an", other: "2"}, object: 2}));
// → false
console.log(deepEqual(obj, {here: {is: "an", other: "3"}, object: 2}));
// → true
```

File: homework/ba-nguyen/lesson-7.2/javascript-3.md [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+```html
+<div onclick="console.log('div')">
+  <p onclick="console.log('p')">Click here!</p>
+</div>
+```
+
+A: p div <br />
+B: div p <br />
+`C: p` => Answer <br />
```
Comment (thaycacac) [inline homework/ba-nguyen/lesson-7.2/javascript-3.md]: Wrong. check answer file

File: homework/ba-nguyen/lesson-7.2/javascript-3.md (hunk 2 above)
Comment (nguyenvanbapc66) [reply homework/ba-nguyen/lesson-7.2/javascript-3.md:120]: I update my code and you can double check again

File: homework/ba-nguyen/lesson-7.2/javascript-3.md (hunk 3 above)
Comment (nguyenvanbapc66) [reply homework/ba-nguyen/lesson-7.2/javascript-3.md]: Oop, i had research before but in some way i choose wrong answer

File: homework/ba-nguyen/lesson-7.2/javascript-3.md (hunk 1 above)
Comment (nguyenvanbapc66) [reply homework/ba-nguyen/lesson-7.2/javascript-3.md:76]: Can you give me a hint? i still don't your mean

Comment (thaycacac) [reply homework/ba-nguyen/lesson-7.2/javascript-3.md:76]: higher-order function need return function
```js
function loop(value, test, update, execute) {
  if (!test(value)) return;
  execute(value);
  return loop(update(value), test, update, execute);
}

loop(3, n => n > 0, n => n - 1, console.log);
```

**Comment by thaycacac** [comment]
LGTM @quanghoang33

**Review by thaycacac — APPROVED** [review]

**Pull request merged**